From ec74615960e534abe0967d40135caba88af8b6f4 Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Tue, 4 Apr 2017 13:20:28 -0700
Subject: [PATCH 1/2] ni: zynq: Mender.io support for NI Ettus Research Sulfur
 Rev3 SDR

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 configs/ni_sulfur_rev3_defconfig |  2 +-
 include/configs/ni_sulfur_rev3.h | 11 ++++-------
 2 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/configs/ni_sulfur_rev3_defconfig b/configs/ni_sulfur_rev3_defconfig
index 2a6a519..42bb18e 100644
--- a/configs/ni_sulfur_rev3_defconfig
+++ b/configs/ni_sulfur_rev3_defconfig
@@ -3,7 +3,7 @@ CONFIG_SYS_VENDOR="ni"
 CONFIG_SYS_CONFIG_NAME="ni_sulfur_rev3"
 CONFIG_ARCH_ZYNQ=y
 CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-sulfur-rev3"
-CONFIG_SYS_EXTRA_OPTIONS="ENV_IS_IN_MMC"
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV,ENV_IS_IN_MMC"
 CONFIG_FIT=y
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/include/configs/ni_sulfur_rev3.h b/include/configs/ni_sulfur_rev3.h
index c10c1f2..3f0be5c 100644
--- a/include/configs/ni_sulfur_rev3.h
+++ b/include/configs/ni_sulfur_rev3.h
@@ -14,19 +14,16 @@
 #include "zynq-common.h"
 #undef CONFIG_ENV_IS_NOWHERE
 
-#define CONFIG_SYS_MMC_ENV_DEV 0
-#define CONFIG_ENV_SECT_SIZE		CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET		0xE0000
-
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-	"fit_image=fit.itb\0"		\
+	"fit_image=boot/fitImage\0"	\
+	"bootargs=root=${mender_kernel_root} rw rootwait\0" \
 	"load_addr=0x2000000\0"		\
 	"fit_size=0x800000\0"           \
 	"fdt_high=0x20000000\0"         \
 	"initrd_high=0x20000000\0"      \
-	"sdboot=echo Copying FIT from SD to RAM... && " \
-		"load mmc 0 ${load_addr} ${fit_image} && " \
+	"sdboot=run mender_setup; echo Copying FIT from SD to RAM... && " \
+		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
 		"tftpboot ${load_addr} ${fit_image} && " \
-- 
2.7.4

